pgmpy/factors/continuous/ContinuousFactor.py

```diff
@@ -0,0 +1,393 @@
+        ...     return (np.power(x, 1)*np.power(y, 2))/beta(x, y)
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> drichlet_factor = ContinuousFactor(['x', 'y'], drichlet_pdf)
+        >>> drichlet_factor.variables
+        ['x', 'y']
+        >>> copy_factor = drichlet_factor.copy()
+        copy_factor.variables
+
+        """
+        return ContinuousFactor(self.scope(), self.pdf)
+
+    def discretize(self, method, *args, **kwargs):
+        """
+        Discretizes the continuous distribution into discrete
+        probability masses using various methods.
+
+        Returns
+        -------
+        An n-D array or a Factor object according to the discretiztion
+        method used.
+
+        Parameters
+        ----------
+        method : A Discretizer Class from pgmpy.discretize
+
+        *args, **kwargs:
+            The parameters to be given to the Discretizer Class.
+        """
+        return method(self, *args, **kwargs).get_discrete_values()
+
+    def reduce(self, values, inplace=True):
+        """
+        Reduces the factor to the context of the given variable values.
+
+        Parameters
+        ----------
+        values: list, array-like
+            A list of tuples of the form (variable_name, variable_value).
+
+        inplace: boolean
+            If inplace=True it will modify the factor itself, else would return
+            a new ContinuosFactor object.
+
+        Returns
+        -------
+        ContinuousFactor or None: if inplace=True (default) returns None
+                                  if inplace=False returns a new ContinuousFactor instance.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from scipy.special import beta
+        >>> def custom_pdf(x, y, z):
+        ...     return z*(np.power(x, 1)*np.power(y, 2))/beta(x, y)
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> custom_factor = ContinuousFactor(['x', 'y', 'z'], custom_pdf)
+        >>> custom_factor.variables
+        ['x', 'y', 'z']
+        >>> custom_factor.assignment(1, 2, 3)
+        24.0
+
+        >>> custom_factor.reduce([('y', 2)])
+        >>> custom_factor.variables
+        ['x', 'z']
+        >>> custom_factor.assignment(1, 3)
+        24.0
+        """
+        if not isinstance(values, (list, tuple, np.ndarray)):
+            raise TypeError("variables: Expected type list or array-like, "
+                             "got type {var_type}".format(var_type=type(values)))
+
+        if not all([var in self.variables for var,value in values]):
+            raise ValueError("Variable not in scope.")
+
+        phi = self if inplace else self.copy()
+
+        var_to_remove = [var for var, value in values]
+        var_to_keep = [var for var in self.variables if var not in var_to_remove]
+
+        reduced_var_index = [(self.variables.index(var), value) for var, value in values]
+        pdf = self.pdf
+
+        def reduced_pdf(*args, **kwargs):
```

Comment:
@yashu-seth Yes. With `types.FunctionType` you will be able to dynamically define functions in a cleaner way.